src/libs/jira-lib.ts

```diff
@@ -0,0 +1,339 @@
+    }
+  }
+  async addCommentToIssueById(issueId: string, comment: string) {
+    if (this.isDryRun) {
+      console.log(`[Dry Run] Would add comment to issue ${issueId}:`, comment);
+      return;
+    }
+
+    try {
+      const response = await this.axiosInstance.post(`/rest/api/2/issue/${issueId}/comment`, { body: comment });
+      await this.removeCurrentUserAsWatcher(issueId); // Commenting on the issue adds the user as a watcher, so we remove them
+
+  
+    } catch (error) {
+      throw new Error(`Error adding comment to issue: ${error}`);
+    }
+  }
+  async findPathToClosure(transitions: any, currentStatus: string) {
+    const visited = new Set();
+    const queue: { path: string[]; status: string }[] = [
+      { path: [], status: currentStatus },
+    ];
+
+    while (queue.length > 0) {
+      const { path, status } = queue.shift()!;
+      visited.add(status);
+
+      const possibleTransitions = transitions.filter(
+        (transition: { from: { name: string } }) =>
+          transition.from.name === status
+      );
+
+      for (const transition of possibleTransitions) {
+        const newPath = [...path, transition.id];
+        const newStatus = transition.to.name;
+
+        if (
+          newStatus.toLowerCase().includes("close") ||
+          newStatus.toLowerCase().includes("done")
+        ) {
+          return newPath; // Found a path to closure
+        }
+
+        if (!visited.has(newStatus)) {
+          queue.push({ path: newPath, status: newStatus });
+        }
+      }
+    }
+
+    return []; // No valid path to closure found
+  }
+
+  async completeWorkflow(issueId: string) {
+    const opposedStatuses = ["canceled", "backout", "rejected"];
+    try {
+      const issue = await this.getIssue(issueId);
+      do {
+        const availableTransitions = await this.getIssueTransitions(issueId);
+        const processedTransitions: string[] = [];
+        console.log(availableTransitions);
+        if (availableTransitions.length > 0) {
+          const targetTransitions = availableTransitions.transitions.filter(
+            (transition: { name: string }) =>
+              !opposedStatuses.includes(transition.name.toLowerCase()) &&
+              !processedTransitions.includes(transition.name.toLowerCase())
+          );
+          const transitionId = targetTransitions[0].id;
+          processedTransitions.push(targetTransitions[0].name);
+          await this.transitionIssue(issueId, {
+            transition: { id: transitionId },
+          });
+          console.log(`Transitioned issue ${issueId} to the next step.`);
+        } else {
+          break;
+        }
+      } while (true);
+    } catch (error) {
+      throw new Error(`Error completing the workflow: ${error}`);
+    }
+  }
+
+  async closeIssue(issueId: string) {
+    if (this.isDryRun) {
+      console.log(`[Dry Run] Would close issue ${issueId}`);
+      return;
+    }
+
+    if (!issueId) return;
```

Comment:
Should we log or throw an error here if `issueId` is empty or undefined?

Reply:
Hmm this looks to be logic pulled over from macpro. I don't seen any need for this to be here at all, why are you calling closed issue with an undefined input in the first place? 

Since we are planning on rewriting the workflow code, lets hold off on changes transition to close related code.

Reply:
👍